### DISCUSSION TITLE
Problem with using GPU for training in Collab

### DISCUSSION THREAD

**Discussion by CarlosQ27** [talmolab/sleap#1699] [Help!]
Hello! I been trying to train a model using Google Collab, but when I try to use the GPU, it doesn't work. I follow the instructions from the website, especifically to run the following two commands:

```bash
!pip uninstall -qqq -y opencv-python opencv-contrib-python
!pip install -qqq "sleap[pypi]>=1.3.3"
```

But when I run them, i get the following error:

```bash
Preparing metadata (setup.py) ... done
Preparing metadata (setup.py) ... done
Building wheel for pykalman (setup.py) ... done
Building wheel for jsmin (setup.py) ... done
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
bigframes 0.21.0 requires scikit-learn>=1.2.2, but you have scikit-learn 1.0.2 which is incompatible.
chex 0.1.85 requires numpy>=1.24.1, but you have numpy 1.22.4 which is incompatible.
fastai 2.7.14 requires pillow>=9.0.0, but you have pillow 8.4.0 which is incompatible.
flax 0.8.1 requires rich>=11.1, but you have rich 10.16.1 which is incompatible.
ibis-framework 7.1.0 requires rich<14,>=12.4.4, but you have rich 10.16.1 which is incompatible.
pandas-gbq 0.19.2 requires google-auth-oauthlib>=0.7.0, but you have google-auth-oauthlib 0.4.6 which is incompatible.
plotnine 0.12.4 requires numpy>=1.23.0, but you have numpy 1.22.4 which is incompatible.
referencing 0.33.0 requires attrs>=22.2.0, but you have attrs 21.4.0 which is incompatible.
tensorflow-datasets 4.9.4 requires protobuf>=3.20, but you have protobuf 3.19.6 which is incompatible.
tensorflow-metadata 1.14.0 requires protobuf<4.21,>=3.20.3, but you have protobuf 3.19.6 which is incompatible.
```

I think that because of some of these errors I am unable to run using a GPU in collab, or I don't know if I'm not taking into account other erros. The problem is that the training takes a lot of time (I am only running 9 epochs with a batch size of 4). Any help would be appreciated. Thanks in advance!

**Comment by Marielamend** [comment]
Hello @CarlosQ27,

Here is the [pinged issue](https://github.com/talmolab/sleap/issues/1644) addressing this problem!
> TLDR: Google Colab no longer works with TensorFlow <2.15.

> Potential workarounds:
Use [Paperspace](https://www.paperspace.com/) as an alternative to Google Colab ...

best,
Mariela

**Comment by galaxiahfast** [comment]
Hello, my friend, did you manage to solve the problem you had? I’m facing something similar.
